src/components/TodoComponent.tsx

```diff
@@ -0,0 +1,280 @@
+            const currentTodo = x;
+
+            if (currentTodo.id === todoId) {
+              currentTodo.completed = !todo.completed;
+            }
+
+            return currentTodo;
+          }));
+        })
+        .catch(() => {
+          setErrorsToFalseExceptTogglingError();
+          setIsTodoToggled(false);
+        });
+    }
+  };
+
+  const onDoubleClick = (i: number) => {
+    setIsTodoEditing(true);
+    setClickedIndex(i);
+  };
+
+  const onEditSubmit = (
+    event: FormEvent<HTMLFormElement>,
+    todoId: number,
+    todoCompleted: boolean,
+  ) => {
+    event.preventDefault();
+    setIsTodoEdited(false);
+
+    const clickedTodoTitle = visibleTodos.find(currentTodo => {
+      return currentTodo.id === todoId;
+    })?.title;
+
+    if (inputRef.current?.value.length === 0 && user) {
+      deleteTodo(user.id, todoId)
+        .then(() => {
+          setVisibleTodos(prev => prev.filter(x => x.id !== todoId));
+          setIsTodoEditing(false);
+          setIsTodoEdited(true);
+        }).catch(() => {
+          inputRef.current?.blur();
+          setIsTodoEditing(false);
+          setIsTodoEdited(true);
+
+          setIsTodoDeleted(true);
+
+          setErrorsToFalseExceptRemoveError();
+        });
+
+      return;
+    }
+
+    if (inputValue === clickedTodoTitle) {
+      inputRef.current?.blur();
+      setIsTodoEdited(true);
+
+      return;
+    }
+
+    if (inputRef.current && user) {
+      updateTodo(user.id, todoId, {
+        title: inputRef.current?.value.trim(),
+        completed: todoCompleted,
+      }).then(() => {
+        setIsTodoEditing(false);
+        setIsTodoEdited(true);
+        setVisibleTodos(prevTodos => prevTodos.map(el => {
+          const prevTodo = el;
+
+          if (prevTodo.id === todoId && inputRef.current) {
+            prevTodo.title = inputRef?.current.value;
+          }
+
+          return prevTodo;
+        }));
+      }).catch(() => {
+        setIsTodoEdited(true);
+        setIsTodoEditing(false);
+
+        setErrorsToFalseExceptTogglingError();
+      });
+    }
+  };
+
+  return (
+    <div
+      key={todo.id}
+      data-cy="Todo"
+      className={classNames(
+        'todo',
+        { completed: todo.completed },
+      )}
+      onDoubleClick={() => {
+        onDoubleClick(index);
+        setInputValue(todo.title);
+        setIsTodoEditing(true);
+        setTimeout(() => {
+          if (inputRef) {
+            inputRef.current?.focus();
+          }
+        }, 0);
+      }}
+    >
+      <label className={classNames(
+        'todo__status-label',
+        { hidden: isTodoEditing && isCurrentClicked },
+      )}
+      >
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          defaultChecked
+          onClick={() => {
+            setClickedIndex(index);
+            handleToggleTodo(todo.id, todo.title);
+          }}
```

Comment:
It's not a good practice to write functions inside JSX. Better to create a separate function for it